### PR TITLE
https://unit.atlassian.net/browse/NP-45350

### PR DESCRIPTION
**Pull request by StigNorland**
promoted papers

### CODE REVIEW

File: search-commons/src/test/java/no/unit/nva/search2/UserSettingsClientTest.java [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+import static nva.commons.core.ioutils.IoUtils.stringFromResources;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+class UserSettingsClientTest {
+
+    private UserSettingsClient userSettingsClient;
+    private static final Logger logger = LoggerFactory.getLogger(ResourceQueryTest.class);
```
[the remark this answers is not in the document]
Comment (StigNorland) [reply search-commons/src/test/java/no/unit/nva/search2/UserSettingsClientTest.java:38]: ups!

File: search-commons/src/main/java/no/unit/nva/search2/UserSettingsClient.java [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    }
+
+    private HttpResponse<String> fetch(HttpRequest httpRequest) {
+        return attempt(() -> httpClient.send(httpRequest, bodyHandler)).orElseThrow();
+    }
+
+    @JacocoGenerated
+    private UserSettings handleResponse(HttpResponse<String> response) {
+        if (response.statusCode() != HTTP_OK) {
+            throw new RuntimeException(response.body());
```
Comment (torarnet) [inline search-commons/src/main/java/no/unit/nva/search2/UserSettingsClient.java:88]: All errors is handled as Internal Server Error? Is this what we want?

**Review by torarnet — COMMENTED** [review]
Overall OK, but have some comments before i approve

File: search-commons/src/main/java/no/unit/nva/search2/ResourceAwsClient.java [hunk 3]
```diff
@@ -161,22 +165,38 @@ private BoolQueryBuilder boolQuery(ResourceAwsQuery query) {
...
         return bq;
     }
 
+    private void addPromotedQuery(ResourceAwsQuery query, BoolQueryBuilder bq) {
+        var promotedPublications = userSettingsClient
+            .doSearch(query)
+            .promotedPublications();
+        for (int i = 0; i < promotedPublications.size(); i++) {
+            bq.should(QueryBuilders.matchQuery("id", promotedPublications.get(i))
+                .boost(promotedPublications.size() - i));
```
[the remark this answers is not in the document]
Comment (StigNorland) [reply search-commons/src/main/java/no/unit/nva/search2/ResourceAwsClient.java:196]: Should is somewhat equal to OR
boost will increase search relevance, and hopefully placement in sort/search
(I have borrowed from old code)

File: search-commons/src/main/java/no/unit/nva/search2/ResourceAwsClient.java [hunk 4]
```diff
@@ -161,22 +165,39 @@ private BoolQueryBuilder boolQuery(ResourceAwsQuery query) {
...
+        }
+        query.removeKey(SORT);
```
Comment (torarnet) [inline search-commons/src/main/java/no/unit/nva/search2/ResourceAwsClient.java]: why is this here?

Comment (StigNorland) [reply search-commons/src/main/java/no/unit/nva/search2/ResourceAwsClient.java]: In the code I borrow from, it doesn't run if sort is set. SORT is always sat in the querybuilder, so I have to remove it in order to make this work, is that is how I interpret the code.

File: search-commons/src/main/java/no/unit/nva/search2/ResourceAwsClient.java [hunk 5]
```diff
@@ -161,22 +165,41 @@ private BoolQueryBuilder boolQuery(ResourceAwsQuery query) {
...
 
+    private void addPromotedQuery(ResourceAwsQuery query, BoolQueryBuilder bq) {
+        var promotedPublications = userSettingsClient
+            .doSearch(query)
+            .promotedPublications();
+        for (int i = 0; i < promotedPublications.size(); i++) {
+            bq.should(QueryBuilders.matchQuery("id", promotedPublications.get(i))
+                .boost(promotedPublications.size() - i));
+        }
+        if (!promotedPublications.isEmpty()) {
```
Comment (torarnet) [inline search-commons/src/main/java/no/unit/nva/search2/ResourceAwsClient.java:198]: extract describing method?

File: search-commons/src/main/java/no/unit/nva/search2/ResourceAwsClient.java (hunk 4 above)
Comment (torarnet) [reply search-commons/src/main/java/no/unit/nva/search2/ResourceAwsClient.java]: maybe that should be described in the code so other people know why? either as describing code or a comment..?

File: search-commons/src/main/java/no/unit/nva/search2/UserSettingsClient.java (hunk 2 above)
Comment (torarnet) [reply search-commons/src/main/java/no/unit/nva/search2/UserSettingsClient.java:88]: or does it return empty list?

**Review by torarnet — DISMISSED** [review]

**Review by torarnet — APPROVED** [review]
